src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -843,10 +885,54 @@ private unsafe void IOLoop()
                     RaiseDataReceivedChars();
                 }
 
+                if (_pinChanged != null)
```

Comment:
If I'm reading this right, we keep pooling the pin status, right? (if somebody is interested) 
Do we have sense how much cpu that burns?

Reply:
When I run Console.ReadLine while this is looping and with Xunit test runner (basically running as one of corefx tests) `top` shows that it oscillates between 0.3-7%. Note we call Thread.Sleep to avoid burning CPU. Considering the other solution is loop but written differently and we need to do looping in order to call DataReceived I think this is fine